### PR TITLE
[SR GeekCamp] Support function array_to_bitmap in StarRocks

### LINKED ISSUE #4010: Support function array_to_bitmap in StarRocks 【Easy】
#  Requirement Description
input: ARRAY_INT ARRAY_SMALLINT ARRAY_TINYINT
output:TYPE_BITMAP
construct a bitmap from array,if input was NULL  then output was null. empty bitmap will be return if array is empty
eg:
array_to_bitmap([1,2,3]) -> bitmap(1,2,3)
array_to_bitmap(NULL) -> NULL
array_to_bitmap([]) -> bitmap()

# Mentor
@stdpain 

# Registration procedure

**Step 1.** Add a comment when you sign up. The comment can be a brief self-introduction and reason for signing up, including what you do, what you're good at, what you're interested in and your technology stack, and how familiar you are with StarRocks. Example:

> I pick! My name is Johnny. I am from StarRocks, and I have been working on database kernel development for one year. I am interested in OLAP, good at C++ programming, and have been using StarRocks for six months. I chose to solve this issue because I am familiar with the execution engine.

**Step 2.** On your WeChat, add the StarRocks community assistant "**_StarRocks-1_**" as a WeChat friend and leave your GitHub ID. The assistant will invite you to a WeChat group based on your picked issue.

**Step 3.** The mentor will communicate with all the participants who picked this issue and assign the task to the best match. 

# Before you start
1. Read the StarRocks [GitHub workflow](https://github.com/StarRocks/community/blob/main/Contributors/guide/workflow.md) carefully, which provides the configurations of BE and FE and describes how to submit a pull request.
2. To learn more about StarRocks, read [StarRocks documentation](https://github.com/StarRocks/docs/blob/main/TOC.md).

# Pull request requirements

1. After you complete code development, upload the code to the StarRocks code repo as a pull request (PR).

  - PR title format: **[SR GeekCamp] + PR title**
  - Labels ：[Geek Camp 2022/feature]
2. After you submit your PR, comment on the status of your PR submission as requested under the Pinned Issue [[StarRocks geek Camp 2022] ](https://github.com/StarRocks/starrocks/issues/4008)page.

  - Comment format: **"I have finished Issue #" + PR link id**

### PR DESCRIPTION
**Pull request by MonsterChenzhuo**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：
close #4010

## Problem Summary(Required) ：
Registered the `array_to_bitmap` function with `bitmapFunctions`.
Implemented the function logic in `BitmapFunctions::array_to_bitmap`

### CODE REVIEW

File: be/src/exprs/vectorized/bitmap_functions.cpp [hunk 1]
```diff
@@ -333,6 +333,43 @@ ColumnPtr BitmapFunctions::bitmap_to_array(FunctionContext* context, const starr
     }
 }
 
+ColumnPtr BitmapFunctions::array_to_bitmap(FunctionContext* context, const starrocks::vectorized::Columns& columns) {
+    size_t size = columns[0]->size();
+    ColumnBuilder<TYPE_OBJECT> builder(size);
+
+    const constexpr PrimitiveType TYPE = TYPE_INT;
+    ArrayColumn* array_column = down_cast<ArrayColumn*>(columns[0]);
```
Comment (stdpain) [inline be/src/exprs/vectorized/bitmap_functions.cpp:341]:
```suggestion
    ArrayColumn* array_column = down_cast<ArrayColumn*>(columns[0].get());
```

**Comment by stdpain** [comment]
add function signature in gensrc/script/vectorized/vectorized_functions.py. you could reference array_length

File: be/src/exprs/vectorized/bitmap_functions.cpp [hunk 2]
```diff
@@ -333,6 +333,43 @@ ColumnPtr BitmapFunctions::bitmap_to_array(FunctionContext* context, const starr
...
+            builder.append_null();
+            continue;
+        }
+        // build bitmap
+        BitmapValue bitmap;
+        for(int j = offset;j<length;j++) {
+            bitmap.add(element_container[j]);
+        }
+        // append bitmap
+        builder.append(&bitmap);
```
Comment (stdpain) [inline be/src/exprs/vectorized/bitmap_functions.cpp:368]:
```suggestion
        builder.append(std::move(bitmap));
```

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/exprs/vectorized/bitmap_functions.cpp [hunk 3]
```diff
@@ -333,6 +333,43 @@ ColumnPtr BitmapFunctions::bitmap_to_array(FunctionContext* context, const starr
     }
 }
 
+ColumnPtr BitmapFunctions::array_to_bitmap(FunctionContext* context, const starrocks::vectorized::Columns& columns) {
+    size_t size = columns[0]->size();
+    ColumnBuilder<TYPE_OBJECT> builder(size);
+
+    const constexpr PrimitiveType TYPE = TYPE_INT;
```
Comment (stdpain) [inline be/src/exprs/vectorized/bitmap_functions.cpp:340]: if input arguments was ARRAY_BIGINT, the type should be TYPE_BIGINT

File: be/test/exprs/vectorized/bitmap_functions_test.cpp [hunk 4]
```diff
@@ -1878,5 +1878,28 @@ TEST_F(VecBitmapFunctionsTest, base64ToBitmapTest) {
     free(src);
 }
 
+TEST_F(VecBitmapFunctionsTest, arrayToBitmapTest) {
+    {
+        Columns columns;
+
+        auto s = BinaryColumn::create();
```
Comment (stdpain) [inline be/test/exprs/vectorized/bitmap_functions_test.cpp:1885]: s shoud be an ArrayColumn

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/exprs/vectorized/bitmap_functions.cpp [hunk 5]
```diff
@@ -333,6 +333,43 @@ ColumnPtr BitmapFunctions::bitmap_to_array(FunctionContext* context, const starr
     }
 }
 
+ColumnPtr BitmapFunctions::array_to_bitmap(FunctionContext* context, const starrocks::vectorized::Columns& columns) {
+    size_t size = columns[0]->size();
+    ColumnBuilder<TYPE_BIGINT> builder(size);
```
Comment (stdpain) [inline be/src/exprs/vectorized/bitmap_functions.cpp:338]:
```suggestion
    ColumnBuilder<TYPE_OBJECT> builder(size);
```

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by stdpain — DISMISSED** [review]

**Review by kangkaisen — DISMISSED** [review]

**Review by Seaven — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by stdpain — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**